### DISCUSSION TITLE
Routes breakpoints

### DISCUSSION THREAD

**Discussion by Emmeron45** [valhalla/valhalla#4065] [Q&A]
Hello! Im new to Valhalla and try to figure out with it's algorithms.

For the test i try to load in Valhalla a lot of gps-points, which are actually can be aggregate in some different routes.

First question: Should I load gps-points in this way? Or i should divide routes from myself and load it to Valhalla.

So when i load multiple routes, Valhalla returns me some map matched routes. Im looked at raw gps data and found breakpoints. I look at Valhalla and found that the routes breaks at approximately same points. 

But i can't understand how it works. I found breakage_distance at https://github.com/valhalla/valhalla/blob/master/docs/docs/meili/configuration.md. But on my routes the distances are much less then default breakage_distance. 

For example, at 1st route driver arrived at side street, waited there about 20 minutes, and then turned around and going in another place. Valhalla gives me a break between this 2 routes, this is exactly what i need, but i cant explain this or maybe i didnt found this at documentation.

So the second question: how the breakpoints working at Valhalla if i load multiple routes, or maybe that was accidentally and i should load one route?

**Comment by kevinkreiser** [comment]
im not 100% certain i understand you but you can control what happens at the points in the map match by using the `location.type` in your request. you'll need to send the request like:

`{"shape":[{"lat":1234,"lon":1234,"type":"via"},...]}`

to read more about the different types of locations we support (4 in total) and what they do see this part of the documentation: https://github.com/valhalla/valhalla/blob/master/docs/docs/api/turn-by-turn/api-reference.md#locations

**Comment by Emmeron45** [comment]
I think that this is not what i wanted. I try to explain in other way.

For example, I got gps tracks from driver for 1 week. The start and the end location in this gps tracks will be different every day. 

So, the 1st question again: can Valhalla consider that there are not only one route and breaks routes by days or load all gps locations is mistake and i should create route for 1 day only 

And since i didnt know the answer for the question, i try send request with all gps points. At the response i get a lot of different routes, so that means that Valhalla can transform gps points into different routes. And i try to understand how it works.

I can understand if start location of current route far away from previous route, the "breakage_distance" indicate to this.
But on my gps points i have different routes with distance much less than default breakage_distance, and Valhalla breaks them.
What is this, the accident? options i didnt found in documentation? or maybe there are mistakes in my request properties and this route should be connected in one?

**Reply by nilsnolde** [reply]
You’re forgetting about time. Don’t know much about Meili or it’s config (also likely that smth is undocumented there), but it sounds quite intuitive to me that Valhalla would break after a certain (maybe adjustable) time interval.

**Reply by Emmeron45** [reply]
Yes, there are time settings, with lon and lat i can load time in seconds, and this surely will affect the result, but for now im not using time. 

And, if i think right, the only time i get is "duration" for every maneuver in json response. But i have no idea how to connect duration and breaks in route. 

Your suggestion about breaks after certain time interval sounds logic if i put time in request. But how it works without my time values?
